mirascope/core/vertex/tool.py

```diff
@@ -0,0 +1,108 @@
+"""The `VertexTool` class for easy tool usage with Google's Vertex LLM calls.
+
+usage docs: learn/tools.md#using-tools-with-standard-calls
+"""
+
+from __future__ import annotations
+
+from typing import Any
+
+from google.cloud.aiplatform_v1beta1.types import (
+    tool as gapic_tool_types,
+)
+from pydantic.json_schema import SkipJsonSchema
+from vertexai.generative_models import (
+    FunctionDeclaration,
+    Tool,
+)
+
+from ..base import BaseTool
+
+
+class VertexTool(BaseTool[Tool]):
+    """A class for defining tools for Vertex LLM calls.
+
+    Example:
+
+    ```python
+    from mirascope.core import prompt_template
+    from mirascope.core.vertex import vertex_call
+
+
+    def format_book(title: str, author: str) -> str:
+        return f"{title} by {author}"
+
+
+    @vertex_call("vertex-1.5-flash", tools=[format_book])
+    @prompt_template("Recommend a {genre} book")
+    def recommend_book(genre: str):
+        ...
+
+
+    response = recommend_book("fantasy")
+    if tool := response.tool:  # returns an `VertexTool` instance
+        print(tool.call())
+    ```
+    """
+
+    __provider__ = "vertex"
+
+    tool_call: SkipJsonSchema[gapic_tool_types.FunctionCall]
```

Comment:
same thing here, why alias this instead of just importing `FunctionCall` directly?

Reply:
I follow the sdk coding style. I change it.

Reply:
Fixed